lib/perl/Genome/SoftwareResult.pm

```diff
@@ -788,7 +788,7 @@ sub delete {
     };
 
     #hook our anonymous sub into the commit callback
-    $observer = $class_name->ghost_class->add_observer(aspect=>'commit', callback=>$upon_delete_callback);
```

Comment:
`$observer` can be removed from this subroutine entirely (in addition to on this line).